### PR TITLE
First pass at M1 support

### PR DESCRIPTION
**Pull request by evertiro**
This is a first-pass at a build working on the Apple M1 (#2385). It mostly works, with a few caveats:

* This is dependent on #2386. There are no 18.x builds for the M1. I'm not building one to support something we shouldn't be using anyway.
* ~~Mailhog still doesn't work. My change to the mailhog provisioner is a bandaid at best. The file I'm linking to is an ARM build, not an ARM64 build. It makes vagrant happy enough to not error out, but won't actually run on M1. I'm looking into building it on the M1.~~ This is both invalid and outdated... we needed aarch64, not arm64. Neither existed. Both are now hosted on my GitHub for `mhsendmail` and `MailHog`.
* It should go without saying, but this assumes that you are using Parallels. This also means that in order to run this, you need to be using Parallels with a Business/Pro license. Once I figure out the mailhog issues, I'll look into VirtualBox on M1.

Since the only two points that this changes only run if you're on an ARM64 machine, this shouldn't impact users on any other architecture.

## Checks

* [x] I've updated the changelog.
* [x] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [x] This PR is complete and ready for review.

### CODE REVIEW

File: provision/core/git/sources-ubuntu-bionic.list [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # git lfs (large file storage plugin for git)
-deb https://packagecloud.io/github/git-lfs/ubuntu/ bionic main
-deb-src https://packagecloud.io/github/git-lfs/ubuntu/ bionic main
```
Comment (tomjn) [inline provision/core/git/sources-ubuntu-bionic.list:3]: This change isn't necessary, and wouldn't apply anyway, a `sources-ubuntu-focal.list` is the solution

File: provision/core/mailhog/provision.sh [hunk 2]
```diff
@@ -10,13 +10,22 @@ function mailhog_setup() {
 
   if [[ ! -e /usr/local/bin/mailhog ]]; then
     vvv_info " * Installing MailHog"
-    curl --silent -L -o /usr/local/bin/mailhog https://github.com/mailhog/MailHog/releases/download/v1.0.0/MailHog_linux_amd64
+
+    if [[ "aarch64" == $(uname -m) ]]; then
+      curl --silent -L -o /usr/local/bin/mailhog https://github.com/evertiro/MailHog/releases/download/v1.0.0-M1/MailHog_linux_arm64
+    else
+      curl --silent -L -o /usr/local/bin/mailhog https://github.com/mailhog/MailHog/releases/download/v1.0.0/MailHog_linux_amd64
+    fi
```
Comment (tomjn) [inline provision/core/mailhog/provision.sh:18]: this could be separated into its own PR, Ubuntu 18 runs on Arm too ( e.g. [Raspberry Pis](https://maker.pro/raspberry-pi/projects/install-ubuntu-18044-lts-on-your-raspberry-pi-board)  )

File: provision/core/mailhog/provision.sh [hunk 3]
```diff
@@ -10,13 +10,22 @@ function mailhog_setup() {
...
     vvv_success " * Mailhog binary installed"
   fi
   if [[ ! -e /usr/local/bin/mhsendmail ]]; then
     vvv_info " * Installing MHSendmail"
-    curl --silent -L -o /usr/local/bin/mhsendmail https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64
+    if [[ "aarch64" == $(uname -m) ]]; then
+      curl --silent -L -o /usr/local/bin/mhsendmail https://github.com/evertiro/mhsendmail/releases/download/v0.2.0-M1/mhsendmail_linux_arm64
+    else
+      curl --silent -L -o /usr/local/bin/mhsendmail https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64
+    fi
```
Comment (tomjn) [inline provision/core/mailhog/provision.sh:28]: see above comment

File: provision/core/vvv/sources-ubuntu-bionic.list [hunk 4]
```diff
@@ -1,3 +1,3 @@
 # VVV mirror packages
-deb http://ppa.launchpad.net/varying-vagrant-vagrants/php/ubuntu bionic main
-deb-src http://ppa.launchpad.net/varying-vagrant-vagrants/php/ubuntu bionic main
+#deb http://ppa.launchpad.net/varying-vagrant-vagrants/php/ubuntu bionic main
+#deb-src http://ppa.launchpad.net/varying-vagrant-vagrants/php/ubuntu bionic main
```
Comment (tomjn) [inline provision/core/vvv/sources-ubuntu-bionic.list:3]: there's a focal source now too, and this file isn't installed on focal anyway

File: provision/core/git/sources-ubuntu-bionic.list (hunk 1 above)
Comment (evertiro) [reply provision/core/git/sources-ubuntu-bionic.list]: I don't know where that one came from...

File: provision/core/mailhog/provision.sh (hunk 2 above)
Comment (evertiro) [reply provision/core/mailhog/provision.sh:18]: Agreed

Comment (evertiro) [reply provision/core/mailhog/provision.sh:18]: This is resolved with #2423

File: provision/core/mailhog/provision.sh (hunk 3 above)
Comment (evertiro) [reply provision/core/mailhog/provision.sh:28]: This is resolved with #2423

**Comment by evertiro** [comment]
The M1-specific stuff from this is now in `develop`, the rest is covered by other PRs.
